src/components/launcheritem.cpp

```diff
@@ -116,6 +139,66 @@ void LauncherItem::setFilePath(const QString &filePath)
     emit this->itemChanged();
 }
 
+namespace {
+
+QDBusInterface* getSailjailInterface()
+{
+    static QDBusInterface *sailjailInterface = NULL;
+    if (!sailjailInterface) {
+        qDBusRegisterMetaType<PropertyMap>();
+        sailjailInterface = new QDBusInterface(
+                SailjailService, SailjailPath, SailjailInterface, QDBusConnection::systemBus());
```

Comment:
Try to avoid using QDBusInterface. On construction it queries the service for its introspection data using blocking calls to construct a dynamic meta object with properties and signals which go unused.

https://code.qt.io/cgit/qt/qtbase.git/tree/src/dbus/qdbusinterface.cpp#n156
https://code.qt.io/cgit/qt/qtbase.git/tree/src/dbus/qdbusintegrator.cpp#n2576

Particularly for a single call you're better of just constructing the method call yourself.
```
QDBusMessage method = QDBusMessage::createMethodCall(SailjailService, SailjailPath, SailjailInterface, SailjailGetAppInfo);
method.setArguments({ sandboxingID() });
QDBusPendingCall async = QDBusConnection::systemBus().asyncCall(method);
```

And for signal connections.

```
QDBusConnection::systemBus().connect(SailjailService, SailjailPath, SailjailInterface, "ApplicationAdded", this, SLOT(sandboxingInfoChanged(QString));
```

Reply:
This is very good to know. I had not realised that it makes such blocking calls. Thanks! I will make the needed changes.

Reply:
Changed.